src/ServiceControl.Audit/Persistence/PersistenceManifest.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace ServiceControl.Audit.Persistence
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+
+    class PersistenceManifest
+    {
+        public string Version { get; set; }
+        public string Name { get; set; }
+        public string DisplayName { get; set; }
+        public string Description { get; set; }
+        public string TypeName { get; set; }
+    }
+
+    static class PersistenceManifestLibrary
+    {
+        public static List<PersistenceManifest> PersistenceManifests { get; set; }
+
+        static bool initialized;
+        static void Initialize()
+        {
+            if (PersistenceManifests == null)
+            {
+                PersistenceManifests = new List<PersistenceManifest>();
+            }
+
+            if (!initialized)
+            {
+                initialized = true;
+                var assemblyLocation = GetEntryOrExecutingAssemblyDirectory();
+                Directory.EnumerateFiles(assemblyLocation, "persistence.manifest", SearchOption.AllDirectories).ToList().ForEach(manifest =>
+                {
+                    PersistenceManifests.Add(System.Text.Json.JsonSerializer.Deserialize<PersistenceManifest>(File.ReadAllText(manifest)));
+                });
+            }
+        }
+
+        static string GetEntryOrExecutingAssemblyDirectory()
+        {
+            var assemblyLocation = Assembly.GetEntryAssembly()?.Location ?? Assembly.GetExecutingAssembly().Location;
+            return Path.GetDirectoryName(assemblyLocation);
+        }
+
+        public static string Find(string persistenceType)
```

Comment:
Same as transports, consider extracting a private method that finds a persistence definition and re-use it for these two methods.